tests/subsys/net/lib/lwm2m/lwm2m_registry/testcase.yaml

```diff
@@ -0,0 +1,6 @@
+common:
+  depends_on: netif
+tests:
+  net.lwm2m.engine:
+    platform_allow: native_posix
```

Comment:
Do we need to limit tests to `native_posix` only? Would be good to allow them to run on other platforms as well.